miner/worker.go

```diff
@@ -343,6 +343,11 @@ func (w *worker) pending() (*types.Block, *state.StateDB) {
 	if w.snapshotState == nil {
 		return nil, nil
 	}
+	snapshotStateCopy := w.snapshotState.Copy()
+	// Call Prepare to ensure that access logs from the previously applied
+	// transaction do not affect gas use calculation for the any subsequently
+	// applied transactions.
+	snapshotStateCopy.Prepare(common.Hash{}, 0)
```

Comment:
Firstly, this `snapshotStateCopy` is not returned, it's noop.

And secondly I don't think it's a good approach to fix it. I would vote for fixing it by reset accessList here https://github.com/ethereum/go-ethereum/blob/master/core/state/statedb.go#L816

Reply:
Well spotted! My original intent was to return the `snapshotStateCopy`  :D

I wasn't convinced by this approach either, I'm definitely in favour of your suggestion, so let's see if it works!